### PR TITLE
Added new capabilities for XCUI 

### PR DESCRIPTION
**Pull request by praveendvd**
## Proposed changes

This pull request will fix https://github.com/webdriverio/webdriverio/issues/7843

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [X] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [X] I have added proper type definitions for new commands (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

File: packages/wdio-types/src/Capabilities.ts [hunk 1]
```diff
@@ -558,6 +558,139 @@ export interface AppiumIOSCapabilities {
     webkitDebugProxyPort?: number;
 }
 
+/**
+ * Appium xcuitest Capabilities
+ *
+ * @see https://github.com/appium/appium-xcuitest-driver
+ */
+export interface AppiumXCUITestCapabilities {
+    'appium:platformName'?:                         string;
```
Comment (christian-bromann) [inline packages/wdio-types/src/Capabilities.ts:567]: Can we remove the spaces and remove the `;` at the end, just so we stay compliant with the rest of the code style, e.g.:
```suggestion
    'appium:platformName'?: string
```

Comment (praveendvd) [reply packages/wdio-types/src/Capabilities.ts:567]: Sure Christian will do that 🎉

**Review by christian-bromann — APPROVED** [review]
Awesome, thanks 👍

**Comment by christian-bromann** [comment]
fyi: the build is broken due to typing issues that I am currently fixing.

**Comment by amandaflagg** [comment]
thank you @praveendvd 💯

**Pull request merged**